src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -187,15 +192,11 @@ public IWebHost Build()
 
         private IServiceCollection BuildHostingServices()
         {
-            // Apply the configuration settings
-            var configuration = _config ?? WebHostConfiguration.GetDefault();
-
-            var mergedConfiguration = new ConfigurationBuilder()
-                                .Add(new IncludedConfigurationProvider(configuration))
-                                .AddInMemoryCollection(_settings)
-                                .Build();
+            if (_config == null)
+            {
+                _config = WebHostConfiguration.GetDefault();
```

Comment:
This is almost never going to execute now because most of the UseSettingFoo will populate _config. Just do `new ConfigurationBuilder().AddInMemoryCollection().Build();` here to avoid unpleasant surprises.